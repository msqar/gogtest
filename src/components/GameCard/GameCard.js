import React from 'react';
import { Card } from 'react-bootstrap';
import PROD_STATUS from '../../utils/prod_status';

class GameCard extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    handleAddToCart = (id) => {
        const { onAddToCart } = this.props;
        if (onAddToCart) onAddToCart(id);
    }

    renderPriceBadge = ({status, price, id}) => {
        switch(status) {
            case PROD_STATUS.NOT_PURCHASED:
                return <button className="Button--small" onClick={() => this.handleAddToCart(id)}>$ {price}</button>
            case PROD_STATUS.IN_CART:
                return <div className="Badge--dark">IN CART</div>
            case PROD_STATUS.OWNED:
                return <div className="Badge--disabled">OWNED</div>
        }
    }

    render() {
        const { info } = this.props;
        return (
            <Card className="GameCard">
                <Card.Img className="GameCard-thumbnail" variant="top" src={info.thumbnail} />
                <Card.Body className="GameCard-body">
                    <Card.Title className="GameCard-title">{info.title}</Card.Title>
                </Card.Body>
                <Card.Footer className="GameCard-footer">
                    {
                        info.discount &&
                        <div className="Badge">-{info.discount}</div>
                    }
                    { this.renderPriceBadge(info) }
                </Card.Footer>
            </Card>
        );
    }
}

export default GameCard;