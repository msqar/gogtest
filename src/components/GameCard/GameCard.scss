@import '../../styles/palette';

.GameCard {
    background-color: $gray-100;
    width: 100%;
    display: flex;
    flex-direction: column;
    border-radius: 0;
    border: none;
    box-shadow: 0 1px 4px 0 rgba(0, 0, 0, .15);

    &:hover {
        box-shadow: 0 3px 4px 0 rgba(0, 0, 0, .15);
    }
}

.GameCard-body {
    padding: 9px;
}

.GameCard-title {
    font-size: 12px;
    line-height: 14px;
    font-weight: bold;
    display: block;
    text-transform: uppercase;
    margin: 5px;
}

.GameCard-thumbnail {
    width: 100%;
    border-radius: 0;
}

.GameCard-footer {
    align-items: flex-end;
    justify-content: flex-end;
    padding: 8px;
    padding-right: 5px;
    display: flex;
    background-color: transparent;
    border: none;
}

@media screen and (max-width: 1000px ) {
    .GameCard {
        margin-bottom: 20px;
    }
}