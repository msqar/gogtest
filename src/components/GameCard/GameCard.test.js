import React from 'react';
import renderer from 'react-test-renderer';
import GameCard from './GameCard';

const mockedItem = {
    thumbnail: '/images/game_thumbnail_1.png',
    price: 9.99,
    title: 'Some Game Title',
    status: 1
};

it('renders GameCard correctly', () => {
    const tree = renderer.create(<GameCard info={mockedItem} />).toJSON();
    expect(tree).toMatchSnapshot();
});