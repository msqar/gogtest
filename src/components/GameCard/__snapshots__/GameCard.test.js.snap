// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GameCard correctly 1`] = `
<div
  className="GameCard card"
>
  <img
    className="card-img-top GameCard-thumbnail"
    src="/images/game_thumbnail_1.png"
  />
  <div
    className="GameCard-body card-body"
  >
    <div
      className="GameCard-title card-title h5"
    >
      Some Game Title
    </div>
  </div>
  <div
    className="GameCard-footer card-footer"
  >
    <button
      className="Button--small"
      onClick={[Function]}
    >
      $ 
      9.99
    </button>
  </div>
</div>
`;
